### PR TITLE
Add OpInfo for torch.baddbmm

### PR DESCRIPTION
**Pull request by anjali411**

### CODE REVIEW

File: test/test_autograd.py [hunk 1]
```diff
@@ -5373,22 +5373,13 @@ def run_functional_checks(test_case, test_name, name, apply_fn, run_grad_checks,
 # NOTE: Some non-holomorphic are separately tested in TestAutogradComplex until gradcheck works properly
 # for non-holomorphic functions
 
-complex_list_filter = []
-
-# TODO: Add back 'sgn' to complex_list; removed because of Windows test failure with 11.2
-# See: https://github.com/pytorch/pytorch/issues/51980
-if _get_torch_cuda_version() != (11, 2):
```
Comment (anjali411) [inline test/test_autograd.py:5380]: cleanup

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -763,6 +763,33 @@ def sample_inputs_addcmul_addcdiv(op_info, device, dtype, requires_grad, **kwarg
...
+def sample_inputs_baddbmm(op_info, device, dtype, requires_grad, **kwargs):
+    test_cases = [((S, S, M), (S, S, S), (S, S, M), 1, 1),
+                  ((1,), (S, S, S), (S, S, M), 1, 1),
+                  ((S, S, M), (S, S, S), (S, S, M), 0.6, 0.2),
+                  ((1,), (S, S, S), (S, S, M), 0.6, 0.2),
+                  ((), (S, S, S), (S, S, M), 1, 1),
+                  ((), (S, S, S), (S, S, M), 0.6, 0.2),
+                  ]
+    sample_inputs = []
+    for input_args in test_cases:
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:775]: style nit: unpacking might have provided nicer names than using indices like `input_args[0]`

File: torch/testing/_internal/common_methods_invocations.py [hunk 3]
```diff
@@ -3090,6 +3117,24 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
...
+           dtypesIfCPU=all_types_and_complex_and(torch.float16, torch.bfloat16),
+           dtypesIfCUDA=floating_types_and(torch.float16, torch.complex64, torch.complex128,
+                                           *[torch.bfloat16] if CUDA11OrLater else []),
+           skips=(
+               # baddbmm does not correctly warn when resizing out= inputs
+               SkipInfo('TestCommon', 'test_out'),
+               # cuda gradchecks are slow
+               # see discussion https://github.com/pytorch/pytorch/pull/47761#issuecomment-747316775
+               SkipInfo('TestGradients', 'test_fn_gradgrad', device_type='cuda'),
+               # Reference for the three skips below: https://github.com/pytorch/pytorch/issues/56508
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:3131]: Does labeling the sample inputs with `broadcasts_input` when there is lhs broadcasting address these issues?

See https://github.com/pytorch/pytorch/blob/b66a1e00a6432c6383e5b1f3f4a61f1ec65a44a4/torch/testing/_internal/common_methods_invocations.py#L100

Comment (anjali411) [reply torch/testing/_internal/common_methods_invocations.py:3131]: yeah that works thanks!

**Comment by mruberry** [comment]
Lint failures are real

I'm not sure what's going on with some of the other failures; want to retrigger CI with the lint fix and see if they're just flaky builds?

**Review by heitorschueroff — APPROVED** [review]
LGTM! The failures seem unrelated. windows builds have been failing on master.

**Pull request closed without merging**